### DISCUSSION TITLE
Can we integrate keygen licence with docker images in private docker repo ?

### DISCUSSION THREAD

**Discussion by yogithakakarla** [keygen-sh/community#137] [General]
we have gone through this link  (https://keygen.sh/blog/how-to-license-and-distribute-a-private-docker-image/)  . but we want to track licence with docker image present on private docker repo ?

if user with proper licence is trying to access (docker pull image) docker image present in prvate docker repo , it should allow , if licence expired it should not allow to pull images

**Comment by ezekg** [comment]
I'm assuming what you're wanting to do is add license validation in front of a private third-party Docker registry. Right now, this isn't supported. But I do have [this issue](https://github.com/keygen-sh/keygen-api/issues/642) open, but it's not a high priority. We eventually [want to offer full OCI support via a release engine](https://github.com/keygen-sh/keygen-api/issues/493).

But until then, you may need to build your own microservice (like https://github.com/keygen-sh/keygen-api/issues/642) that sits in front of your Docker registry, which communicates with Keygen to assert license validity before streaming the files.

Let me know if any of that is incorrect — would love to help get you pointed in the right direction.

**Comment by yogithakakarla** [comment]
Hello,


Thank you for your reply.
Yes, we want to add licence validation to docker image which is present in
any of the docker registry (be it GHCR, Dockerhub, etc)

Our image is pushed in either of this container registries and we want to
add licence validation for users pulling this image from repo.

Hope I made my requirement clear

Thanks,
Yogitha

**Comment by ezekg** [comment]
Just to clarify, because your question is kind of ambiguous… are you asking:

1. How to add a proxy in front of an existing Docker registry that redirects when the user has a valid license.
2. How to add license validation to the Dockerized application itself.

Those are both very different asks, so just want to be on the same page.

**Comment by yogithakakarla** [comment]
Yes. Correct. First point . That's what I am asking for.

Thanks,
Yogitha

**Reply by ezekg** [reply]
Then I'm not sure how my previous response didn't answer your question. It included future plans to eventually tackle the problem, linked to existing issues discussing the problem, and it mentioned that for now you would need to build this yourself on top of Keygen.

**Comment by xXAvoraXx** [comment]
I want to license an application I have developed as monthly, yearly packages. Can I do this with keygen.sh after dockerizing my application?
